docs/architecture/002-event-side-effects.md

```diff
@@ -23,3 +23,42 @@ Pollers should be separate for testability, because we want to
 decouple failures, and perhaps future configurability. We'll of course
 have some share infrastructure code e.g. for storing which events have
 been processed in the DB.
+
+## Amendment: process managers
+
+Date: 2019-11-21
+Authors: @opqdonut @luontola
+
+### Problems with existing pollers
+
+Since one event can generate multiple emails, and only some of these
+emails might fail, we shouldn't retry to process the whole event.
+
+The entitlement poller was causing entitlements to not be immediately
+available after approving an application, which made implementing
+integrations harder.
+
+### New design: process managers and outboxes
+
+A new design is to have _process managers_ that run right after the
+command handler. The process managers look at the events produces by
+the command, and produce a) new commands b) other side-effects.
```

Comment:
Slightly abstract when read for the first time without knowing well in advance what's going on, maybe add a couple of examples of possible side-effects here?

Reply:
It's clearer now, thanks.